### ISSUE TITLE
reportUnnecessaryIsInstance not triggering for "never" cases

### ISSUE THREAD

**Issue by eamartin** [microsoft/pyright#10140]
**Describe the bug**
reportUnnecessaryIsInstance triggers when the isinstance call would always resolve True but not when the isinstance check would always resolve False.

**Code or Screenshots**

```python
# pyright: strict

def f(x: int):
    if isinstance(x, int):
        print("an int")

def g(x: int):
    if isinstance(x, list):
        print("a list")
```

Pyright output:
```
test.py:4:8 - error: Unnecessary isinstance call; "int" is always an instance of "int" (reportUnnecessaryIsInstance)
1 error, 0 warnings, 0 informations
```

Expected output: There should also be a reportUnnecessaryIsInstance for "int is never an instance of list".

**VS Code extension or command-line**
CLI, bug occurs with both v1.1.396 and v1.1.397

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

This isn't considered a "never case". In general, multiple inheritance allows a value to be a subclass of multiple classes.

```pythnon
def g(x: int):
    if isinstance(x, list):
        reveal_type(x) # Subclass of <int and list>
```

In this particular case, it's not possible for a class to subclass from both `int` and `list` for internal layout reasons, but a type checker has no way of knowing this based on the type information provided to it.
